rust/src/builders/tx_builder.rs

```diff
@@ -970,6 +970,26 @@ impl TransactionBuilder {
             total_collateral: self.calc_collateral_total()?,
             reference_inputs: self.reference_inputs.as_ref().map(|inputs| TransactionInputs(inputs.iter().map(|utxo| utxo.input.clone()).collect())),
         };
+
+        built.inputs.0.sort_by(|a, b| {
```

Comment:
nit: maybe add a comment saying why we sort these two fields and only these two fields